base/compiler/abstractinterpretation.jl

```diff
@@ -2316,6 +2316,20 @@ function abstract_call_known(interp::AbstractInterpreter, @nospecialize(f),
     elseif f === UnionAll
         call = abstract_call_gf_by_type(interp, f, ArgInfo(nothing, Any[Const(UnionAll), Any, Any]), si, Tuple{Type{UnionAll}, Any, Any}, sv, max_methods)
         return abstract_call_unionall(interp, argtypes, call)
+    elseif f === _typeof_captured_variable && la == 2
+        t = argtypes[2]
+        if t isa Const
+            tv = t.val
+            rt = Const(has_free_typevars(tv) ? typeof(tv) : Core.Typeof(tv))
+        elseif isType(t)
+            tv = t.parameters[1]
+            rt = Const(has_free_typevars(tv) ? typeof(tv) : Core.Typeof(tv))
```

Comment:
This seems mostly likely incorrect changing an uncertain value to a guaranteed one

Reply:
You mean just the `isType` branch? Or is there an issue with this optimization as a whole?

Reply:
Bump. @vtjnash What would a correct version of this look like?

Reply:
Can you just delete it?